crates/gitbutler-core/src/virtual_branches/branch/mod.rs

```diff
@@ -25,16 +27,10 @@ pub struct Branch {
     pub upstream: Option<git::RemoteRefname>,
     // upstream_head is the last commit on we've pushed to the upstream branch
     pub upstream_head: Option<git::Oid>,
-    #[serde(
-        serialize_with = "serialize_u128",
-        deserialize_with = "deserialize_u128"
-    )]
-    pub created_timestamp_ms: u128,
-    #[serde(
-        serialize_with = "serialize_u128",
-        deserialize_with = "deserialize_u128"
-    )]
-    pub updated_timestamp_ms: u128,
+    #[serde(rename = "created_timestamp_ms", with = "duration_int_string_serde")]
```

Comment:
I think it's a great idea, and can be a template. Instead of putting the unit into the field name on the Rust side, rely on strong (and correct) typing. Then during translation, bake the unit into the field so the frontend has an easier time with units.

Reply:
This particular timestamp is getting persisted into the `virtual_branches.toml` rather than getting sent to the frontend.

I quite like the way Duration was maintaining the same attribute name (so there is less ambiguity between what you see on the frontend and the backend), but provided an object with meaningful keys.